core-tests/src/test/java/org/mule/runtime/core/internal/profiling/tracing/event/span/optel/ExportOnEndCoreEventSpanFactoryTestCase.java

```diff
@@ -163,8 +164,9 @@ public void noExportSpanWhenUsingNoExportUntilSpanCustomizationInfo() {
     InternalSpanExporter spanExporter = ((ExportOnEndSpan) span).getSpanExporter();
     assertThat(spanExporter, instanceOf(OpenTelemetrySpanExporter.class));
     OpenTelemetrySpanExporter openTelemetrySpanExporter = (OpenTelemetrySpanExporter) spanExporter;
-    assertThat(openTelemetrySpanExporter.noExportUntil(), hasSize(1));
+    assertThat(openTelemetrySpanExporter.noExportUntil(), hasSize(2));
```

Comment:
This test was previously wrong. The noExportUntil had to be merged between parent and child.